vnext/ReactUWP/Utils/BaseScriptStoreImpl.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <jsi/jsi.h>
+#include <jsi/ScriptStore.h>
+
+#include <vector>
+#include <algorithm>
+#include <fstream>
+#include <tuple>
+
+namespace react {
+namespace uwp {
+
+struct BufferStore
+{
+  virtual std::unique_ptr<const facebook::jsi::Buffer> getBuffer(const std::string& bufferId) noexcept = 0;
+  virtual bool persistBuffer(const std::string& bufferId, std::unique_ptr<const facebook::jsi::Buffer>) noexcept = 0;
+};
+
+class LocalFileSimpleBufferStore : public BufferStore
+{
+public:
+  LocalFileSimpleBufferStore(const std::string& storeDirectory)
+		: storeDirectory_(storeDirectory) {}
+
+  LocalFileSimpleBufferStore() : storeDirectory_(getApplicationLocalFolder()) {}
+
+  std::unique_ptr<const facebook::jsi::Buffer> getBuffer(const std::string& bufferId) noexcept override;
+  bool persistBuffer(const std::string& bufferId, std::unique_ptr<const facebook::jsi::Buffer>) noexcept override;
+
+private:
+  std::string getApplicationLocalFolder();
+  std::string storeDirectory_;
+};
+
+struct ScriptVersionProvider 
+{
+  virtual facebook::jsi::ScriptVersion_t getVersion(const std::string& url) noexcept = 0;
+};
+
+class LocalFileSimpleScriptVersionProvider : public ScriptVersionProvider
```

Comment:
Did you intend to include the implementation for this class?

Reply:
I think we may start versioning the script source at some point in time .. and it will be useful to keep this code so that we could easily customize it ..